base/range.jl

```diff
@@ -921,14 +921,22 @@ function iterate(r::Union{StepRangeLen,LinRange}, i::Integer=zero(length(r)))
     unsafe_getindex(r, i), i
 end
 
-iterate(r::OrdinalRange) = isempty(r) ? nothing : (first(r), first(r))
-
-function iterate(r::OrdinalRange{T}, i) where {T}
+_iterate(r::OrdinalRange{T}, i::Nothing) where {T} = isempty(r) ? nothing : (first(r), first(r))
+function _iterate(r::OrdinalRange{T}, i) where {T}
     @inline
     i == last(r) && return nothing
     next = convert(T, i + step(r))
     (next, next)
 end
+iterate(r::OrdinalRange, i = nothing) = _iterate(r, i)
+
+# optimized implementation that returns `nothing` for large offsets
+function iterate(r::AbstractOneTo, st = (first(r), zero(step(r))))
```

Comment:
```suggestion
function iterate(r::AbstractOneTo, st = (firstindex(r), zero(step(r))))
```

probably doesn't matter, but just for the intent?